### PR TITLE
Kubernetes: fix process name when the docker registry URL contains a port

### PR DESCRIPTION
**Pull request by rcheatham-q**
## What
This change modifies how the process name is generated to properly handle connectors whose docker registry URL contains a port, e.g. `my-registry:5001/my-project/my-image:0.1.0`

## How
Instead of using the text up until the first colon (via `str.string.split(":")[0]`) the process name generator uses the text up until the last colon (via `str.substring(0, str.lastIndexOf(":"))`.

## 🚨 User Impact 🚨
No significant user impact, unless users have somehow keyed off of the process name Airbyte uses for custom tooling

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Can you add a test to validate that probably with the test you can remove the comment.

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

**Comment by marcosmarxm** [comment]
Hello :wave::skin-tone-2: and thank you for your contribution!

Airbyte has instituted a code freeze between 19 and 30 December, to make sure there are no disruptions during the holidays. 
Because of this, reviewing and merging your contribution may take longer than usual. 
We apologize for the delay, but we want everyone to have a quiet and happy holiday.

If you have any questions or need further clarification, please don't hesitate to ping via Slack.

File: airbyte-workers/src/main/java/io/airbyte/workers/process/ProcessFactory.java [hunk 1]
```diff
@@ -62,7 +62,13 @@ Process create(String jobType,
    * can be used by the factories implementing this interface for easier operations.
    */
   static String createProcessName(final String fullImagePath, final String jobType, final String jobId, final int attempt, final int lenLimit) {
-    final var noVersion = fullImagePath.split(VERSION_DELIMITER)[0];
+    final var delimiterIndex = fullImagePath.lastIndexOf(VERSION_DELIMITER);
+    String noVersion;
+    if (delimiterIndex < 0) {
+      noVersion = fullImagePath;
```
Comment (cgardens) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/ProcessFactory.java:68]: won't the version still be attached?

File: airbyte-workers/src/main/java/io/airbyte/workers/process/ProcessFactory.java [hunk 2]
```diff
@@ -62,7 +62,13 @@ Process create(String jobType,
    * can be used by the factories implementing this interface for easier operations.
    */
   static String createProcessName(final String fullImagePath, final String jobType, final String jobId, final int attempt, final int lenLimit) {
-    final var noVersion = fullImagePath.split(VERSION_DELIMITER)[0];
+    final var delimiterIndex = fullImagePath.lastIndexOf(VERSION_DELIMITER);
+    String noVersion;
+    if (delimiterIndex < 0) {
```
Comment (cgardens) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/ProcessFactory.java:67]: what's the case where this is conditional is true?

**Comment by cgardens** [comment]
Hi @rcheatham-q ! Thanks for your contribution. I left a question about the implementation in the PR. We'd love to get this upstreamed. We've moved our platform code to this [new public repo](https://github.com/airbytehq/airbyte-platform). Would you mind moving you change over there and tagging me? Going to close this PR as the code has moved.

**Pull request closed without merging**